pkg/sources/sources.go

```diff
@@ -38,6 +38,14 @@ type Chunk struct {
 	Verify bool
 }
 
+// GitSourceMetadata defines a common interface for Git-based source metadata.
+// For example, this should match Git, Azure, Bitbucket, GitHub, and Gitlab.
+type GitSourceMetadata interface {
+	GetRepository() string
+	GetCommit() string
+	GetFile() string
+}
+
```

Comment:
I really like this idea, assuming we can find a performant way to actually use it.

Reply:
What are the performance implications of this right now?

Reply:
Ah, you have to cast the `struct` to `interface{}` first.
```
md := data.GetGit()
if d, ok := interface{}(md).(GitSource); ok {
   d.GetFile() // do something
}
```